Short program using arrays fails to compile

### Describe the bug

The following program fails to compile
```
mut a := [][2]int{cap: 10}
a << [0, 0]!
print([0, 0]! in a)
```

The error that is printed is
```
>> v -cg bug.v
C:/Users/user/AppData/Local/Temp/v_0/bug.01JAE5QJDJBAZTECS0SAW0QYZD.tmp.c:529: warning: WINVER redefined
C:/Users/user/AppData/Local/Temp/v_0/bug.01JAE5QJDJBAZTECS0SAW0QYZD.tmp.c:7036: warning: implicit declaration of function 'tcc_backtrace'
C:/Users/user/AppData/Local/Temp/v_0/bug.01JAE5QJDJBAZTECS0SAW0QYZD.tmp.c:13596: error: identifier expected
builder error:
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```

### Reproduction Steps

Try compiling the program I provide above.

### Expected Behavior

The program should compile successfully.

### Current Behavior

The program does not compile successfully.

### Possible Solution

No idea, I just started using V today.

### Additional Information/Context

_No response_

### V version

V 0.4.8 a5adbe8

### Environment details (OS name and version, etc.)

Windows 11 Home
Version 23H2
Build 22631.4317
11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/array.v
+++ b/vlib/v/gen/c/array.v
@@ -1085,13 +1085,19 @@ fn (mut g Gen) gen_array_contains(left_type ast.Type, left ast.Expr, right_type 
 		&& right.info is ast.IdentVar && g.table.sym(right.obj.typ).kind in [.interface, .sum_type]) {
 		g.write('*')
 	}
-	sym := g.table.sym(elem_typ)
-	if sym.kind in [.interface, .sum_type] {
+	if g.table.sym(elem_typ).kind in [.interface, .sum_type] {
 		g.expr_with_cast(right, right_type, elem_typ)
-	} else if right is ast.ArrayInit && sym.info is ast.ArrayFixed {
-		g.write('(${g.typ(right.typ)})')
-		g.fixed_array_var_init(g.expr_string(right), is_auto_deref_var, sym.info.elem_type,
-			sym.info.size)
+	} else if right is ast.ArrayInit {
+		if g.is_cc_msvc {
+			stmts := g.go_before_last_stmt().trim_space()
+			tmp_var := g.new_tmp_var()
+			g.write('${g.typ(right_type)} ${tmp_var} = ${g.expr_string(right)};')
+			g.write(stmts)
+			g.write(tmp_var)
+		} else {
+			g.write('(${g.typ(right.typ)})')
+			g.expr(right)
+		}
 	} else {
 		g.expr(right)
 	}

--- a/vlib/v/gen/c/array.v
+++ b/vlib/v/gen/c/array.v
@@ -1079,16 +1079,19 @@ fn (mut g Gen) gen_array_contains(left_type ast.Type, left ast.Expr, right_type 
 	} else {
 		left_sym.array_fixed_info().elem_type
 	}
-	if (right.is_auto_deref_var() && !elem_typ.is_ptr())
+	is_auto_deref_var := right.is_auto_deref_var()
+	if (is_auto_deref_var && !elem_typ.is_ptr())
 		|| (g.table.sym(elem_typ).kind !in [.interface, .sum_type, .struct] && right is ast.Ident
 		&& right.info is ast.IdentVar && g.table.sym(right.obj.typ).kind in [.interface, .sum_type]) {
 		g.write('*')
 	}
-	if g.table.sym(elem_typ).kind in [.interface, .sum_type] {
+	sym := g.table.sym(elem_typ)
+	if sym.kind in [.interface, .sum_type] {
 		g.expr_with_cast(right, right_type, elem_typ)
-	} else if right is ast.ArrayInit {
+	} else if right is ast.ArrayInit && sym.info is ast.ArrayFixed {
 		g.write('(${g.typ(right.typ)})')
-		g.expr(right)
+		g.fixed_array_var_init(g.expr_string(right), is_auto_deref_var, sym.info.elem_type,
+			sym.info.size)
 	} else {
 		g.expr(right)
 	}

--- a/vlib/v/gen/c/array.v
+++ b/vlib/v/gen/c/array.v
@@ -988,7 +988,7 @@ fn (mut g Gen) gen_array_contains_methods() {
 			fn_builder.writeln('\tfor (int i = 0; i < a.len; ++i) {')
 			if elem_kind == .string {
 				fn_builder.writeln('\t\tif (fast_string_eq(((string*)a.data)[i], v)) {')
-			} else if elem_kind == .array && elem_is_not_ptr {
+			} else if elem_kind in [.array, .array_fixed] && elem_is_not_ptr {
 				ptr_typ := g.equality_fn(elem_type)
 				fn_builder.writeln('\t\tif (${ptr_typ}_arr_eq(((${elem_type_str}*)a.data)[i], v)) {')
 			} else if elem_kind == .function {
@@ -1086,6 +1086,9 @@ fn (mut g Gen) gen_array_contains(left_type ast.Type, left ast.Expr, right_type 
 	}
 	if g.table.sym(elem_typ).kind in [.interface, .sum_type] {
 		g.expr_with_cast(right, right_type, elem_typ)
+	} else if right is ast.ArrayInit {
+		g.write('(${g.typ(right.typ)})')
+		g.expr(right)
 	} else {
 		g.expr(right)
 	}

--- a/vlib/v/tests/builtin_arrays/fixed_array_literal_infix_test.v
+++ b/vlib/v/tests/builtin_arrays/fixed_array_literal_infix_test.v
@@ -1,0 +1,8 @@
+fn test_main() {
+	mut a := [][2]int{}
+	a << [0, 0]!
+	println([0, 0]! in a)
+	ret := [0, 0]! in a
+	println(ret)
+	assert ret
+}

